### PR TITLE
sample 35: one-to-one cascade deletion

### PR DESCRIPTION
**Pull request by gtpan77**
**Reason**
Users has doubt that removing any side of row in a relation would result in deletion of both side of row in the relation. [link](https://stackoverflow.com/questions/64002621/how-to-use-ondelete-cascade-on-one-to-one-relationship/64005071#comment113189983_64005071)

**Solution**
I found that only deleting referencing row would delete referenced row. Otherwise, it will only delete only one side.

If it looks legit, You can also close the issue.

### CODE REVIEW

**Comment by gtpan77** [comment]
@imnotjames, Please review this PR

File: sample/sample35-one-to-one-cascade-deletion/entity/User.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+@Entity()
+export class User {
+
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @Column()
+  name: string;
+
+  @OneToOne(() => Profile, (profile: { user: any; }) => profile.user, { eager: true, onDelete: "CASCADE" })
```
Comment (pleerock) [inline sample/sample35-one-to-one-cascade-deletion/entity/User.ts:13]: why do you need `(profile: { user: any; })` ? why not just `profile => profile.user` ?

**Comment by pleerock** [comment]
can you please add a description to this PR - motivation behind this PR? Thanks.

File: sample/sample35-one-to-one-cascade-deletion/entity/User.ts (hunk 1 above)
Comment (gtpan77) [reply sample/sample35-one-to-one-cascade-deletion/entity/User.ts:13]: Thanks, I'll fix it.

**Comment by gtpan77** [comment]
@pleerock, I have provided the description and motivation for this PR.

**Comment by pleerock** [comment]
I changed your code and tests are still passing - it means you don't need onDelete from both sides. Only from the side where JoinColumn persists. It's a basic behaviour of any RDBMS and relations, it's not really related to ORM.

**Comment by gtpan77** [comment]
Thanks for your change 👍

**Comment by pleerock** [comment]
I don't see enough reasons to merge this new sample. I you had a question, and I think you got your answer now. It will be helpful to others if you answer question on SO. 

p.s. looks like `subscribersDir` change is needed since its a bugfix.

**Comment by gtpan77** [comment]
Sure, You can close this PR. `subscribersDir` change was needed. Now, It is fixed in master branch.

**Comment by pleerock** [comment]
Great, thanks!

**Pull request closed without merging**